examples/src/createReactAgent/accessThreadLevelState.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Access Thread Level State in Message Generation
+ *
+ * This enables the agent to access and utilize the current conversation history and thread-specific state
+ * when crafting responses, maintaining conversational coherence and context awareness.
+ *
+ * Why this is important:
+ * - Conversational Continuity:
+ *   Maintains coherent dialogue by referencing previous exchanges and building upon established context
+ * - Adaptive Responses:
+ *   Modifies response style and content based on conversation flow and user interaction patterns
+ * - State-Aware Decision Making:
+ *   Makes informed choices about tools and actions based on what has already occurred in the conversation
+ *
+ * Example Scenario:
+ * You're building a coding tutor bot. If the user has been asking about Python basics for several messages,
```

Comment:
mmmmm no this isnt greeat

we want an example where the agent in a previous tool call or something updated state (NOT messages key, some other key) and that is now pulled into prompt